### PR TITLE
Fix all necessary sonarcloud issues in workflow resource

### PR DESCRIPTION
**Pull request by garyluu**
For #4349 which is a very specific case that sonarcloud actually detected. 

Decided to fix all legit issues in workflow resource too while i'm at it

### CODE REVIEW

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java [hunk 1]
```diff
@@ -1324,7 +1324,7 @@ public Workflow manualRegister(@ApiParam(hidden = true) @Parameter(hidden = true
                             // Store tool table json
                             toolsJSONTable = lInterface.getContent(w.getWorkflowPath(), mainDescriptor.getContent(),
                                     extractDescriptorAndSecondaryFiles(existingTag), LanguageHandlerInterface.Type.TOOLS, toolDAO);
-                            existingTag.setToolTableJson(toolsJSONTable.get());
+                            toolsJSONTable.ifPresent(existingTag::setToolTableJson);
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java:1327]: Did not know you can do this; I like it.

**Review by NatalieEO — APPROVED** [review]

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java [hunk 2]
```diff
@@ -1384,6 +1384,9 @@ public String getWorkflowDag(@ApiParam(hidden = true) @Parameter(hidden = true,
         checkOptionalAuthRead(user, workflow);
 
         WorkflowVersion workflowVersion = getWorkflowVersion(workflow, workflowVersionId);
+        if (workflowVersion == null) {
+            throw new CustomWebApplicationException("could not find workflow version", HttpStatus.SC_NOT_FOUND);
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java:1388]: Almost all our CustomWebApplicationException messages start with a capital.

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java [hunk 3]
```diff
@@ -1860,6 +1863,9 @@ public Response getWorkflowZip(@ApiParam(hidden = true) @Parameter(hidden = true
...
```
Comment (coverbeck) [inline dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java:1867]: Same comment

**Review by coverbeck — APPROVED** [review]

**Review by denis-yuen — COMMENTED** [review]
A little unclear on the resulting behavior, presumably the `toolsJSONTable` is written to the version properly now. But what is resulting error code when someone tries to snapshot a CommandLineTool registered as a workflow? 

Might be worth a test to lock it down.

File: dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java (hunk 1 above)
Comment (garyluu) [reply dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java:1327]: I did not know either, Intellij suggested it.

**Comment by garyluu** [comment]
> A little unclear on the resulting behavior, presumably the `toolsJSONTable` is written to the version properly now. But what is resulting error code when someone tries to snapshot a CommandLineTool registered as a workflow?
> 
> Might be worth a test to lock it down.

it's a 500 null pointer exception. sonarcloud is the test

**Comment by denis-yuen** [comment]
> it was originally a 500 null pointer exception. now it's just 200. sonarcloud is the test. the problem appears to be when the get-toolsJSONTable returns nothing.

Hmmm, I think the expectation was for an error code of some kind. Just not a 500

>  ` An error should be thrown, but not a 500`

https://github.com/dockstore/dockstore/pull/4287 has some examples of a 400

**Comment by garyluu** [comment]
> > it was originally a 500 null pointer exception. now it's just 200. sonarcloud is the test. the problem appears to be when the get-toolsJSONTable returns nothing.
> 
> Hmmm, I think the expectation was for an error code of some kind. Just not a 500
> 
> > ` An error should be thrown, but not a 500`
> 
> #4287 has some examples of a 400

I edited my comment, it's not 200. It's a variety of codes depending on a bunch of unrelated issues

**Comment by denis-yuen** [comment]
Hmmm, possibly better. What is the error code when someone tries to snapshot a CommandLineTool registered as a workflow? 
I think it still might be worth a test to lock down that error code.

**Comment by garyluu** [comment]
Added the niche test that triggers the original 500. According to the docs, we're allowed to snapshot invalid workflow versions.

**Comment by denis-yuen** [comment]
> Added the niche test that triggers the original 500. According to the docs, we're allowed to snapshot invalid workflow versions.

hmmm, alright but can you link the docs that need to be changed to https://github.com/dockstore/dockstore/issues/4291 so we don't forget?

**Comment by garyluu** [comment]
> > Added the niche test that triggers the original 500. According to the docs, we're allowed to snapshot invalid workflow versions.
> 
> hmmm, alright but can you link the docs that need to be changed to #4291 so we don't forget?

It doesn't look like the docs need to be changed unless we decide to disallow that function (which should be another issue that requires a discussion) https://docs.dockstore.org/en/develop/advanced-topics/snapshot-and-doi.html?highlight=snapshot#what-are-the-requirements-to-snapshot FYI

**Review by denis-yuen — APPROVED** [review]

**Pull request merged**